kube-client/src/client/builder.rs

```diff
@@ -101,8 +100,24 @@ impl TryFrom<Config> for ClientBuilder<BoxService<Request<hyper::Body>, Response
             ));
 
             let mut connector = TimeoutConnector::new(connector);
-            connector.set_connect_timeout(timeout);
-            connector.set_read_timeout(timeout);
+
+            // Set the timeout for the client and fallback to default deprecated timeout until it's removed
+
+            if config.connect_timeout == None {
+                connector.set_connect_timeout(config.timeout);
+            } else {
+                connector.set_connect_timeout(config.connect_timeout);
+            }
```

Comment:
```rust
connector.set_connect_timeout(config.connect_timeout.or(onfig.timeout));
connector.set_read_timeout(config.read_timeout.or(config.timeout));
connector.set_write_timeout(config.write_timeout);
```

Reply:
Thank you, I didn't know that. This is much better